storage/storage.go

```diff
@@ -233,6 +239,73 @@ func NewGRPCClient(ctx context.Context, opts ...option.ClientOption) (*Client, e
 	return &Client{tc: tc}, nil
 }
 
+// CheckDirectConnectivitySupported checks if Direct Connectivity is available
+// for a specific bucket.
+//
+// Implementation currently uses client-side metrics for gRPC to detect
+// Direct Connectivity using grpc.lb.locality label.
+//
+// You may configure the client by passing in options from the [google.golang.org/api/option]
+// package.
+func CheckDirectConnectivitySupported(ctx context.Context, bucket string, opts ...option.ClientOption) (bool, error) {
```

Comment:
I wonder if we should just make this return an error, and nil error means DP worked?

I think we should wrap and return any errors from the metrics stuff to help debug if things go wrong.

Reply:
I like this idea; instead of returning a boolean emit an err when DP isn't available otherwise no error means DP is available.